### PR TITLE
4255 Update bundle download

### LINKED ISSUE #4255: Update Bundle download function
The bundle hosting place has been defined, thus the download function should be updated.

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Signed-off-by: Yiheng Wang <vennw@nvidia.com>

Fixes #4255 .

### Description
This PR is used to update the default repo of bundle download and load functions. In addition, it also enhances the load function.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/bundle/scripts.py [hunk 1]
```diff
@@ -293,7 +300,7 @@ def load(
     configer = ConfigComponent(config=net_kwargs)
     model = configer.instantiate()
     model.to(device)  # type: ignore
-    model.load_state_dict(model_dict)  # type: ignore
+    copy_model_state(dst=model, src=model_dict if key_in_ckpt is None else model_dict[key_in_ckpt])  # type: ignore
```
Comment (wyli) [inline monai/bundle/scripts.py:303]: do you think this util should be used somewhere in this function? https://github.com/Project-MONAI/MONAI/blob/2bd56dc7b35422a60fbd39bed38e094c423a986b/monai/networks/utils.py#L383-L394

File: monai/bundle/scripts.py [hunk 2]
```diff
@@ -247,15 +253,16 @@ def load(
         model_file: the relative path of the model weights or TorchScript module within bundle.
             If `None`, "models/model.pt" or "models/model.ts" will be used.
         load_ts_module: a flag to specify if loading the TorchScript module.
-        bundle_dir: the directory the weights/TorchScript module will be loaded from.
+        bundle_dir: directory the weights/TorchScript module will be loaded from.
             Default is `bundle` subfolder under`torch.hub get_dir()`.
-        source: the place that saved the bundle.
+        source: place that saved the bundle.
```
Comment (ericspod) [inline monai/bundle/scripts.py:258]: It's not clear what's meant by this, may be "storage location name" and state that "github" is currently the only available value. This should be updated where this argument is used as well.

Comment (yiheng-wang-nv) [reply monai/bundle/scripts.py:258]: Thanks Eric, I updated the docstrings.

File: monai/bundle/scripts.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/bundle/scripts.py:303]: Hi @wyli ,

I think this PR refers to: https://github.com/Project-MONAI/MONAI/blob/dev/monai/bundle/scripts.py#L592.
The `copy_model_state()` API leverages the `get_state_dict` as you mentioned.

Thanks.

File: monai/bundle/scripts.py [hunk 3]
```diff
@@ -202,7 +200,13 @@ def download(
 
     _log_input_summary(tag="download", args=_args)
     name_, bundle_dir_, source_, repo_, url_, progress_ = _pop_args(
-        _args, name=None, bundle_dir=None, source="github", repo=None, url=None, progress=True
+        _args,
+        name=None,
+        bundle_dir=None,
+        source="github",
+        repo="Project-MONAI/model-zoo/hosting_storage_v1",
```
Comment (Nic-Ma) [inline monai/bundle/scripts.py:207]: I think maybe no need to set default here as you already set in the `__init__()`?

Thanks.

File: monai/bundle/scripts.py [hunk 4]
```diff
@@ -151,7 +151,7 @@ def download(
     name: Optional[str] = None,
     bundle_dir: Optional[PathLike] = None,
     source: str = "github",
-    repo: Optional[str] = None,
+    repo: str = "Project-MONAI/model-zoo/hosting_storage_v1",
```
[the remark this answers is not in the document]
Comment (yiheng-wang-nv) [reply monai/bundle/scripts.py:154]: Thanks Nic, I changed the descriptions.

File: monai/bundle/scripts.py (hunk 1 above)
Comment (wyli) [reply monai/bundle/scripts.py:303]: sure, in that case calling `model_dict[key_in_ckpt]` here might be unstable, because it's assuming a dictionary?

Comment (yiheng-wang-nv) [reply monai/bundle/scripts.py:303]: Thanks @wyli . As defined in https://docs.monai.io/en/latest/mb_specification.html `model.pt` is " the state dictionary of a saved model". Could we just add some more descriptions about it?

Comment (wyli) [reply monai/bundle/scripts.py:304]: I think this is confusing, I'd suggest:

```py
model_dict = torch.load...
if not isinstance(model_dict, Mapping):
    warnings.warn(f"the state dictionary from {full_path} should be a dictionary but go {type(model_dict}.")
    model_dict = get_state_dict(model_dict)

```

**Review by wyli — APPROVED** [review]

File: monai/bundle/scripts.py [hunk 5]
```diff
@@ -281,19 +281,22 @@ def load(
 
     if device is None:
         device = "cuda:0" if is_available() else "cpu"
-    # loading with `torch.jit.load`
+    # loading with `torch.jit. load`
```
Comment (Nic-Ma) [inline monai/bundle/scripts.py:284]: May I know why added a blank here? Is it a typo?

Thanks.

Comment (yiheng-wang-nv) [reply monai/bundle/scripts.py:284]: Thanks Nic, it seems the code format tool mis-modify this place, I just pushed a new commit.

**Review by Nic-Ma — APPROVED** [review]

**Pull request merged**